client/landing/stepper/declarative-flow/internals/steps-repository/importer/hooks/use-step-navigator.ts

```diff
@@ -0,0 +1,90 @@
+import { addQueryArgs } from 'calypso/lib/route';
+import { useCheckoutUrl } from 'calypso/signup/steps/import-from/hooks/use-checkout-url';
+import { WPImportOption } from 'calypso/signup/steps/import-from/wordpress/types';
+import { getWpOrgImporterUrl } from 'calypso/signup/steps/import/util';
+import { StepPath } from '../../../steps-repository';
+import { BASE_STEPPER_ROUTE } from '../../import/config';
+import { removeLeadingSlash } from '../../import/util';
+import type { NavigationControls } from '../../../types';
+import type { StepNavigator } from 'calypso/signup/steps/import-from/types';
+
+export function useStepNavigator(
+	navigation: NavigationControls,
+	siteId: number,
+	siteSlug: string,
+	fromSite: string
+): StepNavigator {
+	const checkoutUrl = useCheckoutUrl( siteId as number, siteSlug as string );
```

Comment:
I'm very curious about all the uses of `as` for type casting in this PR. Why are they here?

In this case, `siteId` and `siteSlug` are already guaranteed to be a number and a string, respectively. There are other cases where the type casting is explicitly ignoring `undefined` values as if they will never exist. Doing this breaks the safety granted by TypeScript and opens the way for fatal errors that will be very difficult to find.

In general, if a value cannot be undefined, we should change where the value comes from rather than type casting. If a value can be undefined but will be ignored by a function or component, we should change the implementation to support undefined as a value. I'm happy to help with specifics if you have any questions!

Reply:
I'm going to replace some of these in https://github.com/Automattic/wp-calypso/pull/66729

Reply:
Hey @sirbrillig,

I agree with you 100% that type casting can potentially cause an issue, but I'm not sure why is there.
Probably, in development time by doing this we temporarily fix type issue and forget to properly fix it later.

Thank you for letting us know, and I'll try to pass through the codebase and fix if there are more unnecessary type casting.